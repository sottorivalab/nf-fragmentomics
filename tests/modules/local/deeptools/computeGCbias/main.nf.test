nextflow_process {

    name "Test Process COMPUTEGCBIAS"
    script "modules/local/deeptools/computeGCbias/main.nf"
    process "COMPUTEGCBIAS"

    test("Should run COMPUTEGCBIAS without failures") {

        when {
            params {                
                outdir = "tests/results"
            }
            process {
                """
                input[0] = [
                    [ caseid:'test', sampleid: 'test', timepoint: 'T1' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.2bit', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.freq, process.out.versions).match()
        }

    }

}
