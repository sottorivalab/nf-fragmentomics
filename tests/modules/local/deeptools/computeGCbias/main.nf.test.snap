{
    "Should run COMPUTEGCBIAS without failures": {
        "content": [
            [
                [
                    {
                        "caseid": "test",
                        "sampleid": "test",
                        "timepoint": "T1"
                    },
                    "test.paired_end.sorted.bam:md5,76396678f283516b159a1e9b538fc64c",
                    "test.paired_end.sorted.bam.bai:md5,9ac6ecb9435822c44b37765f111e9013",
                    "chr22_23800000-23980000.2bit:md5,eb3e744156457a9e644f346493da8b8b",
                    "test.freq.txt:md5,4a7eb3e068f25d4f021ca3740f30fe36"
                ]
            ],
            [
                "versions.yml:md5,116827051220feaa1149750b722b27a2"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-22T10:58:26.202369"
    }
}